src/java/com/rapleaf/jack/queries/where_operators/Match.java

```diff
@@ -1,8 +1,16 @@
 package com.rapleaf.jack.queries.where_operators;
 
+import com.google.common.base.Preconditions;
+import com.rapleaf.jack.queries.Column;
+
 public class Match extends WhereOperator<String> {
 
   public Match(String pattern) {
     super("LIKE ?", pattern);
   }
+
+  public Match(Column<String> column, String prefix, String suffix) {
+    super(String.format("LIKE CONCAT('%s', %s, '%s')", prefix, column.getSqlKeyword(), suffix));
+    Preconditions.checkNotNull(column);
```

Comment:
I just realized that the current way we do `null` check is incorrect in a lot of the matchers. If `column` is `null`, an NPE will have been thrown in the `super` call before reaching this statement.

`checkNotNull` returns the object itself, so it is better to call it in the `super`:
```
super(..., prefix, Preconditions.checkNotNull(column).getSqlKeyword()), suffix));
```